internal/appsec/dyngo/instrumentation/httpinstr/tags.go

```diff
@@ -0,0 +1,91 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package httpinstr
+
+import (
+	"encoding/json"
+	"sort"
+	"strings"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+// setEventSpanTags sets the security event span tags into the service entry span.
+func setEventSpanTags(span ddtrace.Span, events json.RawMessage) {
+	// Set the appsec event span tag
+	// eventTag is the structure to use in the `_dd.appsec.json` span tag.
+	type eventTag struct {
+		Triggers json.RawMessage `json:"triggers"`
+	}
+	// TODO(Julio-Guerra): avoid serializing the json in the request hot path
+	event, err := json.Marshal(eventTag{Triggers: events})
+	if err != nil {
+		log.Error("appsec: unexpected error while serializing the appsec event span tag: %v", err)
+		return
+	}
+	span.SetTag("_dd.appsec.json", string(event))
+	// Keep this span due to the security event
+	span.SetTag(ext.ManualKeep, true)
+	span.SetTag("_dd.origin", "appsec")
+	// Set the appsec.event tag needed by the appsec backend
+	span.SetTag("appsec.event", true)
+}
+
+// setEventSpanTags sets the AppSec-specific span tags when a security event occurred into the service entry span.
+func setSpanTags(span ddtrace.Span, events json.RawMessage, remoteIP string, headers map[string][]string) {
+	setEventSpanTags(span, events)
+	span.SetTag("network.client.ip", remoteIP)
+	for h, v := range normalizeHTTPHeaders(headers) {
+		span.SetTag("http.request.headers."+h, v)
+	}
+}
```

Comment:
Looking for comments on this.

There's a key difference with the previous appsecevt intake implementation: those tags are set in the request handler now. I checked the span implementation and the fact tags need to be `string` values during the `SetTag()` call makes impossible to make it lazy and done during the msgpack serialization only.

I think we're good to go this way for now, but worth keeping in mind for the next iterations and improvements to come, especially once we add request blocking, where the faster you stop an attack request, the better.